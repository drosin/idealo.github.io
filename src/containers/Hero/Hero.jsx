import React, { Component } from 'react';
import './Hero.scss';

import Socials from '../../components/Socials/Socials'
import ProgressiveImage from '../../components/ProgressiveImage/ProgressiveImage';

export default (props) => {
  function scrollDown() {
    let navHeight = document.querySelector('nav').offsetHeight;
    let elemOffset = document.querySelector('#home').getBoundingClientRect().bottom + window.pageYOffset

    window.scrollTo(0, elemOffset - navHeight);
  }

  return (
    <header className='hero__container' id='home'>
      <ProgressiveImage className='hero__image' src='static/work.jpg' thumbnail='static/thumbnails/work.jpg' />

      <div className='hero__imageContainer'>
        <div className='hero__bottomContainer'>
          <p className='hero__bottomText'>There‘s more to it.</p>
          <Socials className='hero__socials' />
        </div>
        {/*<img className='hero__scrollDown' src='static/arrow.png' onClick={scrollDown} />*/}
      </div>
      <p className='hero__text'>
        Nowadays the largest source of revenue at <strong>Axel Springer</strong> and it’s companies is the digital business. Hence, sharing of best practices and practical solutions plays a huge role in our everyday work. That’s why we’re going public with some of our projects here. <strong>We love the idea of cooperating inhouse and with everyone outside who can benefit from the things we’ve built.</strong> Even better if you join us in discussion and development!
      </p >
    </header >
  );
}