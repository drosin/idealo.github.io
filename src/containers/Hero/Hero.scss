@import '../../styles/variables.scss';

.hero{

    &__container {
        height: 100vh;
        display: flex;
        padding-top: 51px;
        flex-direction: column;
    }

    &__imageContainer {
        position: relative;
        z-index: 0;
        flex: 0;
    }
    
    &__image {
        flex: 1;
    }

    &__bottomContainer {
        flex: 0;
        position: absolute;
        bottom: 0;
        width: 100%;
        left: 0;
        display: flex;
        padding: 20px;

        align-items: flex-end;
    }

    &__bottomText {
        font-size: 38px;
        font-weight: bold;
        color: $font-color-secondary;
        line-height: 1.2;
        margin-right: auto;

        @media screen and (max-width : 500px) {
            line-height: 1;
            font-size: 20px;
        }
    }

    &__socials {
        font-size: 24px;
        flex-shrink: 0;

        > a {
            padding: 0 10px;
            transition: font-size .2s;
            line-height: 1;
            
            &:hover {
                font-size: 40px;
                color: #fff;
            }
        }

        @media screen and (max-width : 500px) {
            display: none;
        }
    }   

    &__text {
        font-size: 20px;
        padding: 40px 70px;
        text-align: center;
        background-color: #fff;

        flex: none;

        > strong {
            font-weight: bold;
        }

        @media screen and (max-width : 500px) {
            font-size: 18px;
            padding: 20px 30px;
        }
    }

    &__scrollDown {
        height: 25px;
        position: absolute;
        
        z-index: 3;
        bottom: 20px;
        left: 50%;
        transform: translateX(-50%);  

        animation: jump 4s infinite;

        cursor: pointer;

        @media screen and (max-width : 500px) {
            display: none;
        }
    }
}

@keyframes jump {
    0%, 10%, 20%, 100% {
        transform: translateX(-50%) translateY(-10px)
    }
    5%, 15% {
        transform: translateX(-50%) translateY(0px)
    }
}