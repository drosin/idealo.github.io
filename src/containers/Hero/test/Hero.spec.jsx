import React from 'react';
import renderer from 'react-test-renderer';
import Hero from '../Hero';

jest.mock('../../../components/ProgressiveImage/ProgressiveImage', () => 'ProgressiveImage')

describe('Hero component', () => {
    it('should render as expected', () => {
        const renderedComponent = renderer.create(
            <Hero />
        );
        expect(renderedComponent.toJSON()).toMatchSnapshot();
    })
});