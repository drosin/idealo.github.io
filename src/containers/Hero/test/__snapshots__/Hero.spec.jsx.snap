// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component should render as expected 1`] = `
<header
  className="hero__container"
  id="home"
>
  <ProgressiveImage
    className="hero__image"
    src="static/work.jpg"
    thumbnail="static/thumbnails/work.jpg"
  />
  <div
    className="hero__imageContainer"
  >
    <div
      className="hero__bottomContainer"
    >
      <p
        className="hero__bottomText"
      >
        There‘s more to it.
      </p>
      <p
        className="socials__container hero__socials"
      >
        <a
          className="socials__item"
          href="https://www.xing.com/companies/axelspringerse"
        >
          <span
            aria-hidden={true}
            className="fa fa-xing"
          />
        </a>
        <a
          className="socials__item"
          href="https://de.linkedin.com/company/axel-springer-se"
        >
          <span
            aria-hidden={true}
            className="fa fa-linkedin"
          />
        </a>
        <a
          className="socials__item"
          href="https://www.instagram.com/axel_springer_karriere/"
        >
          <span
            aria-hidden={true}
            className="fa fa-instagram"
          />
        </a>
        <a
          className="socials__item"
          href="https://www.facebook.com/AxelSpringerKarriere/"
        >
          <span
            aria-hidden={true}
            className="fa fa-facebook"
          />
        </a>
      </p>
    </div>
  </div>
  <p
    className="hero__text"
  >
    Nowadays the largest source of revenue at 
    <strong>
      Axel Springer
    </strong>
     and it’s companies is the digital business. Hence, sharing of best practices and practical solutions plays a huge role in our everyday work. That’s why we’re going public with some of our projects here. 
    <strong>
      We love the idea of cooperating inhouse and with everyone outside who can benefit from the things we’ve built.
    </strong>
     Even better if you join us in discussion and development!
  </p>
</header>
`;
